shapeshift/web
# feat: filter down DeFi section assets by wallet support

## Description

Does what it says on the box - note we can't do that at selectors-level since we need the wallet to do such checks. We could theoretically use the first AccountId as a way to achieve similar checks, but due to the way the selectors are (we won't always have a `ChainId` handy i.e when aggregating by provider) we can't use the `selectFirstAccountIdByChainId` selector.

## Pull Request Type


## Issue (if applicable)



closes #5130

## Risk



Low performance risks as we're filtering at component-level, as well as a small risk of positions not displaying when they should

## Testing



- Connect native wallet
- Ensure all/wallet DeFi positions are showing no regression against develop/prod

- Connect MM
- Ensure only EVM chains are shown

- Connect Keplr
- Ensure only Cosmos opportunities are shown

- Ensure all above works both for "By Provider" and by asset
- Ensure there are no stale opportunities while switching from one wallet to another (Native to MM/Keplr is a good way to double check this)

### Engineering



### Operations



## Screenshots (if applicable)


<img width="1473" alt="Screenshot 2023-08-22 at 15 00 23" src="https://github.com/shapeshift/web/assets/17035424/107cd91b-4b1f-415f-8e86-85001defd491">
<img width="1486" alt="Screenshot 2023-08-22 at 15 11 27" src="https://github.com/shapeshift/web/assets/17035424/678b6cc2-2224-49c0-bd3c-3d19e611722c">

<img width="1469" alt="image" src="https://github.com/shapeshift/web/assets/17035424/0fb48da2-5ae3-4af4-be65-ec35aa78c271">
<img width="1456" alt="image" src="https://github.com/shapeshift/web/assets/17035424/f754e028-ea2e-41df-8196-de52fe21a746">

<img width="1472" alt="image" src="https://github.com/shapeshift/web/assets/17035424/ab747016-5ed2-4228-9d6e-d4a94d2f1747">
<img width="1479" alt="image" src="https://github.com/shapeshift/web/assets/17035424/9cdcfa7a-f11c-48b1-bf43-cdc48a1b5e3d">
